Index: main.c
===================================================================
--- main.c	(revision 43092)
+++ main.c	(working copy)
@@ -657,7 +657,8 @@
 
 	if (Area_cat[i].count > 0 && method_opt->answer) {
 	    /* get stats */
-	    statsvalue(&result, Area_cat[i].values, Area_cat[i].nvalues);
+	    statsvalue(&result, Area_cat[i].values, Area_cat[i].nvalues,
+			NULL);
 
 	    if (half)
 		result += 0.5;
