Index: wxgui.py
===================================================================
--- wxgui.py	(revision 47502)
+++ wxgui.py	(working copy)
@@ -1234,14 +1234,14 @@
         Debug.msg(1, "GMFrame.NewDisplay(): idx=%d" % self.disp_idx)
         try:
             from gui_modules.wms.wmsmenu import DisplayWMSMenu
+	    DisplayWMSMenu()
         except:
             gcmd.GError(parent = self.parent,
                         message = _("Experimental WMS support for wxGUI not available. "
                                     "You can install it by '%s'") % \
                             "g.extension -s extension=wx.wms")
             return
-        
-    	DisplayWMSMenu()
+        return
     
     def OnNewDisplay(self, event = None):
         """!Create new layer tree and map display instance"""
