Index: gui/wxpython/wxgui.py
===================================================================
--- gui/wxpython/wxgui.py	(revision 46749)
+++ gui/wxpython/wxgui.py	(working copy)
@@ -1,5 +1,5 @@
 """!
-@package wxgui.py
+@package wxgui.py 
 
 @brief Main Python app for GRASS wxPython GUI. Main menu, layer management
 toolbar, notebook control for display management and access to
@@ -18,7 +18,7 @@
 @author Jachym Cepicky (Mendel University of Agriculture)
 @author Martin Landa <landa.martin gmail.com>
 @author Vaclav Petras <wenzeslaus gmail.com> (menu customization)
-"""
+""" 
 
 import sys
 import os
@@ -81,6 +81,7 @@
 from gui_modules import gmodeler
 from gui_modules import vclean
 from gui_modules import nviz_tools
+from gui_modules.wmsmenu import DisplayWMSMenu #sudeep
 from gui_modules.debug    import Debug
 from gui_modules.ghelp    import MenuTreeWindow, AboutWindow, InstallExtensionWindow
 from gui_modules.toolbars import LMWorkspaceToolbar, LMDataToolbar, LMToolsToolbar, LMMiscToolbar
@@ -1176,10 +1177,18 @@
         # show ATM window
         dbmanager.Show()
         
+    def OnNewDisplay1(self, event = None):
+        """!Create new layer tree and map display instance"""
+        self.NewDisplay1()
+
+    def NewDisplay1(self, show = True):
+        Debug.msg(1, "GMFrame.NewDisplay(): idx=%d" % self.disp_idx)
+    	DisplayWMSMenu()
+    	
     def OnNewDisplay(self, event = None):
         """!Create new layer tree and map display instance"""
         self.NewDisplay()
-
+       
     def NewDisplay(self, show = True):
         """!Create new layer tree, which will
         create an associated map display frame
Index: gui/wxpython/gui_modules/toolbars.py
===================================================================
--- gui/wxpython/gui_modules/toolbars.py	(revision 46749)
+++ gui/wxpython/gui_modules/toolbars.py	(working copy)
@@ -270,6 +270,8 @@
         icons = Icons['displayWindow']
         return self._getToolbarData((('displaymap', icons['display'],
                                       self.parent.OnDraw),
+                                      ('displaymap', icons['display'],
+                                      self.parent.OnDraw),
                                      ('rendermap', icons['render'],
                                       self.parent.OnRender),
                                      ('erase', icons['erase'],
@@ -1472,6 +1474,8 @@
         icons = Icons['layerManager']
         return self._getToolbarData((('newdisplay', icons["newdisplay"],
                                       self.parent.OnNewDisplay),
+                                      ('newdisplay', icons["newdisplay"],
+                                      self.parent.OnNewDisplay1),
                                      (None, ),
                                      ('workspaceNew', icons["workspaceNew"],
                                       self.parent.OnWorkspaceNew),
